plotting/README.md

```diff
@@ -0,0 +1,45 @@
+## Plotting
+
+This directory contains the core MaCh3 plotting tools. `plottingUtils` contains all the code defining the plotting library, the majority of which basically consists of a highly flexible root file reader which can be configured by different experiments to read from their MaCh3 output files and put everything in a nice common format that can then be accessed in plotting applications and allows for easy iteration over parameters and samples. It is also possible to configure the plotting library to be able to read files from other fitters, which is especially useful for cross-fitter validations (a task that tends to take up a lot of time during an analsis). The aim here is to avoid having to have vast numbers of un-maintained plotting scripts for dealing with every possible combination of fitter and analysis iteration and instead maintain this core code and allow experiments to simply maintain their own config files.
+
+## Structure
+
+The plotting library consists of 3 main manager classes: `PlottingManager`, `inputManager` and `styleManager` each of which is responsible for different aspects of plot making.
+
+### PlottingManager
+
+This is intended to be the main class that the user interacts with. It is responsible for parsing user specified options, keeping track of which config files to use, and managing instances of the other manager classes.
+
+### inputManager
+
+This is responsible for keeping track of the files whose data is to be plotted and reading in that data from them. It will attempt to automatically detect which fitter the file came from based on the config-defined file structures. It will read in the data accordingly, storing in a common format which it provides functions to easily obtain.
+
+### styleManager 
+
+This is responsible for the styling of the plots produces. It contains the functionality to apply user specified styles, colour palettes and fancy names for parameters and samples.
+
+## Config Files
+
+The goal with MaCh3s plotting library is to be as flexible as possible and to abstract away all of the most annoying parts about plot making like reading in the data from fitter output files and keeping track of parameter names across different fitters, while allowing the user as much freedom as possible when it comes plot style and formatting . In order to do this, MaCh3s plotting library makes heavy use of config files so that the behaviour of the code can be adapted easily to individual experiments without having to totally reinvent the wheel and rewrite large ammounts of boilerplate code. In line with the rest of MaCh3, the plotting library uses [YAML](https://yaml.org/spec/1.2.2/) to specify its config files. There are a few different config files which control different parts of the plotting library. Some example yaml configs found in this directory.
```

Comment:
Can you add reference to this here: https://github.com/mach3-software/MaCh3/tree/feature_Plotting?tab=readme-ov-file#plotting-and-diagnostic